src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AddMissingInterpolationToken.cs

```diff
@@ -0,0 +1,107 @@
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class AddMissingInterpolationTokenAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA2017";
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+            new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.AddMissingInterpolationTokenTitle), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
+            new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.AddMissingInterpolationTokenMessage), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.IdeSuggestion,
+            description: null,
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterOperationAction(context =>
+            {
+                var literalOperation = (ILiteralOperation)context.Operation;
+
+                if (!literalOperation.ConstantValue.HasValue ||
+                    literalOperation.ConstantValue.Value is not string stringText)
+                {
+                    return;
+                }
+
+                var parts = GetInterpolationParts(stringText);
+                if (parts.All(s => IsValidInterpolationPart(s, literalOperation.SemanticModel, literalOperation.Syntax.SpanStart)))
+                {
+                    context.ReportDiagnostic(literalOperation.CreateDiagnostic(Rule));
+                }
+            }, OperationKind.Literal);
+        }
+
+        private static bool IsValidInterpolationPart(string s, SemanticModel semanticModel, int position)
+        {
+            if (uint.TryParse(s, out _))
+            {
+                // Numerical literals are valid interpolation from language perspective.
+                // But we don't want the analyzer to flag for them. So,
+                // they're invalid from analyzer perspective.
+                return false;
+            }
+            return !semanticModel.LookupSymbols(position, name: s).IsDefaultOrEmpty;
+        }
+
+        private static IEnumerable<string> GetInterpolationParts(string s)
```

Comment:
I think you want to reuse the parsing regex from the [CA2243 PR](https://github.com/dotnet/roslyn-analyzers/pull/4226/files#diff-822f292292d15a89b293bf2c681da69718067c9bfb2c12968980921dee24b1f9R166)

Reply:
@Evangelink Will the regex you used detect nesting?

Reply:
Not sure honestly, it will need some testing. Tagging @jmarolf in case he has a better idea for this handling.